src/main/java/edu/cornell/kfs/fp/service/impl/ScheduledAccountingLineServiceImpl.java

```diff
@@ -0,0 +1,79 @@
+package edu.cornell.kfs.fp.service.impl;
+
+import static org.kuali.kfs.sys.document.validation.impl.AccountingDocumentRuleBaseConstants.ERROR_PATH.DOCUMENT_ERROR_PREFIX;
+
+import java.sql.Date;
+import java.util.Calendar;
+import java.util.TreeMap;
+
+import org.kuali.rice.core.api.util.type.KualiDecimal;
+import org.kuali.rice.krad.util.GlobalVariables;
+
+import edu.cornell.kfs.fp.businessobject.ScheduledSourceAccountingLine;
+import edu.cornell.kfs.gl.service.ScheduledAccountingLineService;
+import edu.cornell.kfs.sys.CUKFSKeyConstants;
+
+public class ScheduledAccountingLineServiceImpl implements ScheduledAccountingLineService {
+	
+	public TreeMap<Date, KualiDecimal> generateDatesAndAmounts(ScheduledSourceAccountingLine scheduledAccountingLine, int rowId) {
+		TreeMap<Date, KualiDecimal> datesAndAmounts = new TreeMap<Date, KualiDecimal>();
+		int count = Integer.parseInt(scheduledAccountingLine.getPartialTransactionCount());
+		Date endDate = generateEndDate(scheduledAccountingLine);
+		KualiDecimal totalAmount = calculateTotalAmount(scheduledAccountingLine.getAmount());
+		Date currentStartDate = (Date) scheduledAccountingLine.getStartDate().clone();
+		int i = 0;
+		while (currentStartDate.before(endDate) && totalAmount.isGreaterThan(KualiDecimal.ZERO)) {
+			Calendar calendarCalculator = Calendar.getInstance();
```

Comment:
this is better, but it still feels like it could be refactored more to pull out some of the date logic here